src/Server/Server.ts

```diff
@@ -59,11 +58,6 @@ export class Server extends BaseServer {
   constructor(params: IConstructorParams = { hostname: "", ip: createRandomIp() }) {
     super(params);
 
-    // "hacknet-node-X" hostnames are reserved for Hacknet Servers
```

Comment:
You can't remove this code. I'm not confident I've tracked down *all* the places the constructor gets called from (it's a lot), but at minimum this can be invoked with old savefiles that have purchased servers named "hacknet-node-X", due to the big you are fixing or other reasons.

Reply:
As far as I can tell, the `Server` class is only constructed by two functions: [`safelyCreateUniqueServer()`](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Server/ServerHelpers.ts#L41) and [`initForeignServers()`](https://github.com/bitburner-official/bitburner-src/blob/dev/src/Server/AllServers.ts#L156). The later will never touch a purchased server so it's not a concern here, and the former is only called in [four places](https://github.com/search?q=repo%3Abitburner-official%2Fbitburner-src%20path%3A%2F%5Esrc%5C%2F%2F%20safelyCreateUniqueServer&type=code). That doesn't seem like "a lot" to me?

Reply:
Also that's a good catch, I'm assuming that it should just create a random hostname like it did before?

Reply:
Yeah, just revert this part and keep it for safety. I like the more explicit checks you added elsewhere, so this should only be needed as a fallback.

Reply:
Reverted via force-push.